services/galley/test/unit/Test/Galley/Mapping.hs

```diff
@@ -0,0 +1,209 @@
+
+  let locals = [mkInternalMember alice]
+  let remotes = []
+  conv <- mkInternalConv locals remotes
+  let actual = cnvMembers <$> conversationViewMaybeQualified convDomain requester conv
+
+  assertEqual "members:" Nothing actual
+
+runMappingRemoteUser :: HasCallStack => IO ()
+runMappingRemoteUser = do
+  let aliceDomain = Domain "backendA.example.com"
+  let convDomain = Domain "backendA.example.com"
+  let bobDomain = Domain "backendB.example.com"
+  alice <- flip Qualified aliceDomain <$> randomId
+  bob <- flip Qualified bobDomain <$> randomId
+  let expectedSelf = Just $ mkMember alice
+  let expectedOthers = Just [mkOtherMember bob]
+
+  let locals = [mkInternalMember alice]
+  let remotes = [mkRemoteMember bob]
+  conv <- mkInternalConv locals remotes
+  let actual = cnvMembers <$> conversationViewMaybeQualified convDomain alice conv
+
+  assertEqual "self:" expectedSelf (cmSelf <$> actual)
+  assertEqual "others:" expectedOthers (cmOthers <$> actual)
+
+runMappingRemoteConv :: HasCallStack => IO ()
+runMappingRemoteConv = do
+  let aliceDomain = Domain "backendA.example.com"
+  let convDomain = Domain "backendB.example.com"
+  let bobDomain = Domain "backendB.example.com"
+  alice <- flip Qualified aliceDomain <$> randomId
+  bob <- flip Qualified bobDomain <$> randomId
+  let expectedSelf = Just $ mkMember alice
+  let expectedOthers = Just [mkOtherMember bob]
+
+  let locals = [mkInternalMember bob]
+  let remotes = [mkRemoteMember alice]
+  conv <- mkInternalConv locals remotes
+  let actual = cnvMembers <$> conversationViewMaybeQualified convDomain alice conv
+
+  assertEqual "self:" expectedSelf (cmSelf <$> actual)
+  assertEqual "others:" expectedOthers (cmOthers <$> actual)
+
+-- Here we expect the conversationView to return nothing, because Alice (the
+-- requester) is not part of the conversation (Her unqualified UUID is part of
+-- the conversation, but the function should catch this possibly malicious
+-- edge case)
+runMappingSameUnqualifiedUUID :: HasCallStack => IO ()
+runMappingSameUnqualifiedUUID = do
+  let aliceDomain = Domain "backendA.example.com"
+  let convDomain = Domain "backendB.example.com"
+  let bobDomain = Domain "backendB.example.com"
+  uuid <- randomId
+  let alice = Qualified uuid aliceDomain
+  let bob = Qualified uuid bobDomain
+
+  let locals = [mkInternalMember bob]
+  let remotes = []
+  conv <- mkInternalConv locals remotes
+  let actual = cnvMembers <$> conversationViewMaybeQualified convDomain alice conv
+
+  assertEqual "members:" Nothing actual
+
+--------------------------------------------------------------
+
+mkOtherMember :: Qualified UserId -> OtherMember
+mkOtherMember u = OtherMember u Nothing roleNameWireAdmin
+
+mkRemoteMember :: Qualified UserId -> RemoteMember
+mkRemoteMember u = I.RemoteMember (toRemote u) roleNameWireAdmin
+
+mkInternalConv :: [LocalMember] -> [RemoteMember] -> IO Data.Conversation
+mkInternalConv locals remotes = do
+  creator <- randomId
```

Comment:
Shouldn't the creator always be part of the conversation? I'd always expect the creator to be on the list of local members (I suppose one cannot create a conversation on a remote backend). Furthermore, given that it is generated at random, we know nothing how it relates to IDs in `locals`, e.g., maybe it is equal to an ID in the list.

Even if the implicit idea behind `Data.Conversation` is that `Data.convCreator` is always a local member (which I'd have to check), I'd turn the creator into an input to `mkInternalConv` to make it more visible at use sites that there is another member of the conversation.

Reply:
The creator has no role in the unit test I'm try to do here, which is about the conversationView. So any value is fine.

A creator often is part of the localMembers; yes; but in theory a creator can create a conversation, then leave the conversation, leading to a situation where he/she is no longer part of that conversation. I'll add a comment as it may be confusing for the reader.